react-native-hub-app/src/checklist.js

```diff
@@ -0,0 +1,449 @@
+          const id = await this.generateIdentity();
+
+          // Pull the stored context to reuse if available && valid date
+          const existingCtx = await this.getContext();
+          if (existingCtx) {
+            const database = new Client(existingCtx);
+            updatedState = {
+              ctx: existingCtx,
+              db: database,
+            };
+          } else {
+            // Signs the API secret to create a new API token that will expire
+            ctx = new Context(API_URL);
+            const sig = await createAPISig(USER_API_SECRET);
+            ctx = ctx.withAPIKey(USER_API_KEY).withAPISig(sig);
+
+            // Create a new crypto identity for your user
+            // The API will return a token to be used for this user
+            const database = new Client(ctx);
+            const token = await this.getUserToken(database, id);
+
+            // We want to update our Context with the token now.
+            ctx = ctx.withToken(token);
+
+            // store the context for future use
+            const ctxStr = JSON.stringify(ctx.toJSON());
+            await AsyncStorage.setItem(CONTEXT_KEY, ctxStr);
+
+            updatedState = {
+              ctx: ctx,
+              db: database,
+              steps: steps,
+            };
+          }
+
+          data.status = 2;
+          steps[stepNumber] = data;
+          this.setState(updatedState);
+
+          break;
+        }
+        case 1: {
+          const tid = await this.getAstronautThread(db);
+          // Update our app state with success
+          data.status = 2;
+          steps[stepNumber] = data;
+          this.setState({
+            threadId: tid,
+            steps: steps,
+            message: ctx,
+          });
+          break;
+        }
+        case 2: {
+          // Create a new instance of an Astronaut
+          const ids = await db.create(threadId, 'Astronaut', [
+            createAstronaut(),
+          ]);
+
+          // Update our app state with success
+          data.status = 2;
+          steps[stepNumber] = data;
+          data.message = ids[0];
+          this.setState({
+            entityId: ids[0],
+            steps: steps,
+          });
+          break;
+        }
+        case 3: {
+          // Search for an Instance with firstName of Buzz
+          const q = new Where('firstName').eq('Buzz');
+          const r = await db.find(threadId, 'Astronaut', q);
+
+          // Update our app state with success (if we really found the instance)
+          data.status = r.instancesList[0]._id === this.state.entityId ? 2 : 9;
+          data.message = r.instancesList[0];
+          steps[stepNumber] = data;
+          this.setState({
+            steps: steps,
+          });
+          break;
+        }
+        case 4: {
+          const buckets = new Buckets(ctx);
+          const roots = await buckets.list();
+          // Update our app state with success (if we really found the instance)
```

Comment:
got a good precanned bucket push that should work without a filesystem?

Reply:
Does RN have a Files API? Here’s a possible example: https://github.com/textileio/js-textile/blob/master/src/buckets.spec.ts#L93

Reply:
Actually, if its just a test, you can also just use:

```
{ path: string, content: UInt8Array/Buffer }
```

See: https://github.com/textileio/js-textile/blob/master/src/normalize.ts#L20